### PR TITLE
[EBPF] Avoid rebuilds and improve kmt.prepare speed

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR improves the `kmt.prepare` task to avoid unnecessary rebuilds and improve speed:

* The `go_package_dirs` is changed to use a single invocation of `go list`, which makes that call slightly faster (it takes several seconds usually).
* Calculate the dependencies of Go test binaries using `go list`, and communicate that to Ninja so that tests are only rebuilt when needed.

### CODE REVIEW

File: tasks/kmt.py [hunk 1]
```diff
@@ -707,6 +707,33 @@ def build_object_files(ctx, fp):
...
+    for line in res.stdout.split("\n"):
+        if ":" not in line:
+            continue
+
+        pkg, deps = line.split(":", 1)
+        deps = [d.strip() for d in deps.split(" ")]
+        dd_deps = [d[len(dd_pkg_name) :] for d in deps if d.startswith(dd_pkg_name)]
+
+        # Test packages might be named path/to/pkg [path/to/pkg.test] or
+        # path/to/pkg.test. Control for both cases and get the package name
```
Comment (val06) [inline tasks/kmt.py:730]: not sure i understand the examples in this comment

File: tasks/kmt.py [hunk 2]
```diff
@@ -773,19 +806,22 @@ def kmt_sysprobe_prepare(
...
-            nw.build(
-                inputs=[pkg],
-                outputs=[output_path],
-                implicit=go_files,
-                rule="gotestsuite",
-                pool="gobuild",
-                variables=variables,
-            )
+            go_files = set(glob(f"{pkg}/*.go"))
+            has_test_files = any(x.endswith("_test.go") for x in go_files)
```
Comment (val06) [inline tasks/kmt.py:810]: maybe use  tolower as well?

**Review by val06 — CHANGES_REQUESTED** [review]
reviewed

File: tasks/kmt.py (hunk 1 above)
Comment (gjulianm) [reply tasks/kmt.py]: I changed the comment to have a better explanation, it's management for the different ways the `ImportPath` property for a test package might be shown by go list, let me know what you think.

File: tasks/kmt.py (hunk 2 above)
Comment (gjulianm) [reply tasks/kmt.py]: Done!

**Review by val06 — APPROVED** [review]

**Pull request merged**